src/domain/animal_detection_mode.py

```diff
@@ -41,7 +41,7 @@ def run(self):
                     self.camera_thread.append(camera.run())
                 elif camera.type == Camera.CameraTypes.FTPCamera and FTPsServer.ftps_server and not self.ftp_thread:
                     logger.info("Instantiating FTPS server...")
-                    FTPsServer.ftps_server.run()
+                    self.ftp_thread = FTPsServer.ftps_server.run()
```

Comment:
Do you define / initialized self.ftp_thread somewhere in the class (__init__)?

Reply:
As this is a specialization class of OperationMode class, self.ftp_thread = None is performed in OperationMode init()